specification/eventgrid/data-plane/Microsoft.DataBox/stable/2018-01-01/DataBox.json

```diff
@@ -0,0 +1,113 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "version": "2018-01-01",
+    "title": "Schema of Azure Data Box events published to Azure Event Grid",
+    "description": "Describes the schema of the Azure Data Box events published to Azure Event Grid. This corresponds to the Data property of an EventGridEvent."
+  },
+  "paths": {},
+  "definitions": {
+    "DataBoxCopyStartedEventData": {
+      "description": "Schema of the Data property of an EventGridEvent for a Microsoft.DataBox.CopyStarted event.",
+      "type": "object",
+      "properties": {
+        "serialNumber": {
+          "description": "Serial Number of the device associated with the event, separated by comma delimiter in case of multiple",
+          "type": "string"
+        },
+        "stageDetails": {
+          "description": "Metadata related to the stage detail",
+          "type": "object",
+          "properties": {
+            "name": {
+              "type": "string",
+              "description": "Name of the current Stage",
+              "enum": [
+                "CopyStarted",
+                "CopyCompleted",
+                "OrderCompleted"
+              ],
+              "x-ms-enum": {
+                "name": "name",
+                "modelAsString": true
+              }
+            },
+            "time": {
+              "description": "The time at which the stage happened.",
+              "format": "date-time",
+              "type": "string"
+            }
+          }
+        }
+      }
+    },
+    "DataBoxCopyCompletedEventData": {
+      "description": "Schema of the Data property of an EventGridEvent for a Microsoft.DataBox.CopyCompleted event.",
+      "type": "object",
+      "properties": {
+        "serialNumber": {
+          "description": "Serial Number of the device associated with the event, separated by comma delimiter in case of multiple",
+          "type": "string"
+        },
+        "stageDetails": {
+          "description": "Metadata related to the stage detail",
+          "type": "object",
+          "properties": {
+            "name": {
```

Comment:
Should this be something like "stage"?

Reply:
Since we are moving with complex object strategy, this field's context can be inferred by it's parent object. so keeping the field as " name" itself, but changing the enum type , just like you mentioned.